source/game_sa/Cover.cpp

```diff
@@ -118,16 +123,37 @@ void CCover::FindCoverPointsForThisBuilding(CBuilding* building) {
 
 // 0x698D40
 uint8 CCover::FindDirFromVector(float x, float y) {
-    return plugin::CallAndReturn<uint8, 0x698D40, float, float>(x, y);
+    return (uint8)((atan2(-x, y)) * 40.743664);
 }
 
 // 0x698D60
 CVector CCover::FindVectorFromDir(uint8 direction) {
-    return plugin::CallAndReturn<CVector, 0x698D60, uint8>(direction);
+    CVector vector;
+    vector.x = (float)sin(direction * 0.02454369);
+    vector.y = (float)cos(direction * 0.02454369);
+    vector.z = 0.0;
```

Comment:
```suggestion
    vector.x = -sinf(direction * 0.02454369f);
    vector.y = cosf(direction * 0.02454369f);
    vector.z = 0.0f;
```
How do you like the idea of ​​adding the following functions to common.h and using them?:

```cpp
constexpr inline uint8_t CompressRad(float rad) { return (uint8_t)(rad * ((256.0f / 2.0f) / PI)); }
constexpr inline float DecompressRad(uint8 rad) { return (float)rad * (PI / (256.0f / 2.0f)); }
```


Then it can be implemented like this:

```cpp
    return CVector{
        -std::sinf(DecompressRad(direction)),
        std::cosf(DecompressRad(direction)),
        0.0f
    };
```

Reply:
> How do you like the idea of ​​adding the following functions to common.h and using them?

`FixedFloat<uint8, 128.0f / PI>` can be used for this purpose

Reply:
We could also typedef it for this purpose.
`CompressedRad`

Reply:
that's kinda errnous tho, it should be 255 / PI... Though seemingly they did use 256 instead, which might've lead to overflows... :D